docs/porting-guide.rst

```diff
@@ -2480,12 +2483,11 @@ Function : plat\_crash\_console\_flush
 ::
 
     Argument : void
-    Return   : int
+    Return   : void
```

Comment:
I see, although only flush() is updated not putc().

The return codes are there generally to follow the stdio.h equivalent functions. I can see there's not much value in returning error codes from the crash functions but I'm not very comfortable with discarding this information from all the underlying drivers, in case they are used by other top-level APIs (e.g. stdlib clients).

In the case of the multi-console API, you could make a policy decision on what to do, for example try outputting on all consoles even on error, and report the first error you come across.

In any case, this change in error code handling is separate to the main part of this patch and it would be easier to review if it were in a separate patch (you might be detecting a theme to my comments here!).